Sources/NIOPosix/Bootstrap.swift

```diff
@@ -521,6 +521,200 @@ public final class ServerBootstrap {
+    /// SwiftNIO extras package. The `ServerQuiescingHelper` can be installed using the
+    /// ``ServerBootstrap/serverChannelInitializer`` callback.
+    ///
+    /// Below you can find the code to setup a simple TCP echo server that supports graceful server closure.
+    ///
+    /// ```swift
+    /// let quiesce = ServerQuiescingHelper(group: group)
+    /// let signalSource = DispatchSource.makeSignalSource(signal: SIGINT, queue: signalQueue)
+    /// signalSource.setEventHandler {
+    ///     signalSource.cancel()
+    ///     print("\nreceived signal, initiating shutdown which should complete after the last request finished.")
+    ///
+    ///     quiesce.initiateShutdown(promise: fullyShutdownPromise)
+    /// }
+    /// try await ServerBootstrap(group: self.eventLoopGroup)
+    ///    .serverChannelInitializer { channel in
+    ///        channel.eventLoop.makeCompletedFuture {
+    ///            try channel.pipeline.syncOperations.addHandler(quiesce.makeServerChannelHandler(channel: channel))
+    ///        }
+    ///    }
+    ///    .serverChannelOption(.socketOption(.so_reuseaddr), value: 1)
+    ///    .bind(
+    ///        target: .hostAndPort(self.host, self.port),
+    ///        childChannelInitializer: { channel in
+    ///            channel.eventLoop.makeCompletedFuture {
+    ///                try channel.pipeline.syncOperations.addHandler(ByteToMessageHandler(NewlineDelimiterCoder()))
+    ///                try channel.pipeline.syncOperations.addHandler(MessageToByteHandler(NewlineDelimiterCoder()))
+    ///
+    ///                return try NIOAsyncChannel(
+    ///                    wrappingChannelSynchronously: channel,
+    ///                    configuration: NIOAsyncChannel.Configuration(
+    ///                        inboundType: String.self,
+    ///                        outboundType: String.self
+    ///                    )
+    ///                )
+    ///            }
+    ///        }
+    ///    ) { channel in
+    ///        print("Handling new connection")
+    ///        await self.handleConnection(channel: channel)
+    ///        print("Done handling connection")
+    ///    }
+    ///
+    /// ```
+    ///
+    /// - Parameters:
+    ///   - target: The ``BindTarget`` to use.
+    ///   - serverBackPressureStrategy: The back pressure strategy used by the server socket channel.
+    ///   - childChannelInitializer: A closure to initialize the channel. The return value of this closure is used in the `onConnection`
+    ///                              closure.
+    ///   - onConnection: A closure to handle the connection. Use the channel's `inbound` property to read from
+    ///                   the connection and channel's `outbound` to write to the connection.
+    ///
+    /// - Note: If the server is not closed using a closure mechanism like above, the bind method must be cancelled using task
```

Comment:
This sentence confused me for a second since I read `closure mechanism` as in the Swift closures but I think you mean the mechanism that is used to close the channel via quiescing. Maybe we can rephrase this as `The bind method respects task cancellation which will force close the server. If you want to gracefully shut-down use the quiescing helper as outlined above`?